### DISCUSSION TITLE
AG Grid find filter data

### DISCUSSION THREAD

**Discussion by snowbollaanm** [zauberzeug/nicegui#2158] [Ideas / Feature Requests]
Hi! 

I've created a function to be put into the AG Grid that will return a dictionary with the values from it's filter. It also contains options to return values from a specific columns filter or all columns. 

```py
    async def get_filter_data(self, colID=None) -> List[Dict]:
        """Get the current filters for a column
        
        This method retunrs the current filters applied to an AG Grid 

        :param colID: an optional parameter for if you want to only return the value of one of the filters
        
        :return: If a colID is parsed returns a dict and if it isn't will return list of dicts 
        """
        result = await self.call_api_method('getFilterModel')

        if colID:
            try:
                return {'type': colID, 'filter': result[colID]['filter']}
            except:
                return {'type': 'No filter found', 'filter': ''}
        else:
            try:
                return [{'type': key, 'filter': value['filter']} for key, value in result.items()]
            except:
                return [{'type': 'No filter found', 'filter': ''}]
```

This can be called one of two ways: 

1.  For returning all filters: `v = await circuit_grid.get_filter_data()`
2. For returning filters of a specific column: `v = await circuit_grid.get_filter_data(colID='column field')`

I was hoping that it could be added to the aggrid py file for the next release.

Thanks!

**Comment by falkoschindler** [comment]
Thanks for the suggestion, @snowbollaanm! We're always looking for ways to improve NiceGUI's usability.

But I'm not sure about `get_filter_data`. Here are some concerns about your code:

- Returning a hard-coded value `{'type': 'No filter found', 'filter': ''}` as an indicator for "no filter" feels a bit unexpected. I think you should return `None` or an empty list, respectively.
- Blindly catching _all_ exceptions (even `KeyboardInterrupt` and `StopIteration`) is bad practice. We should catch more specific exceptions or explicitly check if `colID` is in `result`.
- Using `type` as key for the column ID is a bit strange, because it represents the filter type in the `result` dictionary. If in doubt, we should stick with AG Grid's vocabulary.

I understand that you're trying to find a shortcut to simplify user code. But this way you're moving many specific API decisions and complexity into the library. Right now I'd argue that `await self.call_api_method('getFilterModel')` is good enough and we can leave further processing to the user.
